modules/core/04-channel/keeper/packet_test.go

```diff
@@ -656,6 +656,31 @@ func (suite *KeeperTestSuite) TestAcknowledgePacket() {
 
 			channelCap = suite.chainA.GetChannelCapability(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
 		}, true},
+		{"success on channel in tryupgrade flush status in flushing", func() {
+			// setup uses an UNORDERED channel
+			suite.coordinator.Setup(path)
+
+			// create packet commitment
+			sequence, err := path.EndpointA.SendPacket(defaultTimeoutHeight, disabledTimeoutTimestamp, ibctesting.MockPacketData)
+			suite.Require().NoError(err)
+
+			// create packet receipt and acknowledgement
+			packet = types.NewPacket(ibctesting.MockPacketData, sequence, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, defaultTimeoutHeight, disabledTimeoutTimestamp)
+			err = path.EndpointB.RecvPacket(packet)
+			suite.Require().NoError(err)
+
+			channelCap = suite.chainA.GetChannelCapability(path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+
+			// Move channel to correct state.
+			path.EndpointA.ChannelConfig.ProposedUpgrade.Fields.Version = ibcmock.UpgradeVersion
+			path.EndpointB.ChannelConfig.ProposedUpgrade.Fields.Version = ibcmock.UpgradeVersion
```

Comment:
super nit: i _think_ this line is not strictly necessary as we will fetch the version from the counterparty upgrade from EndpointA when calling `ChanUpgradeTry`on chainB, the channel config proposed upgrade is then only used for the timeout and connection hops.

Reply:
had to double check myself. By default, Endpoint's handlers will get a proposed upgrade populated by what the channel is set up with. We override to specifically get passed the `DeepEqual` check when validating the new upgrade fields against the old ones.

Reply:
Ah, I meant just the version setting for Endpoint B, the endpoint A line still makes sense

Reply:
ah, gotcha. yeah for the non-ChanUpgradeInit caller, setting it is not required, danke. I think we've got some instances of this in other places too.